scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-cli-func.sh.j2

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+set -eux
+set -o pipefail
+
+# This script runs barbican client functional tests from upstream
+
+# Discover cli tests
+export OS_TEST_PATH="{{ ardana_qe_tests_dir }}/functionaltests"
+export OS_TOP_LEVEL="{{ ardana_qe_tests_dir }}"
+export VENV_FOLDER="{{ ardana_qe_test_venv }}"
+export TESTR="{{ ardana_qe_test_venv }}/bin/testr"
+export PYTHON="{{ ardana_qe_test_venv }}/bin/python"
+export TESTS_TO_SKIP_BIN="{{ ardana_qe_tests_dir }}/tests2skip.py"
+export RUN_FILTER="{{ ardana_qe_tests_dir }}/run_filters/${1:-{{ run_filter }}}.txt"
+export REQUESTS_CA_BUNDLE={{ ca_bundle }}
+
+set +u
+source $VENV_FOLDER/bin/activate
+set -u
+cd {{ ardana_qe_tests_dir }}
+
+$VENV_FOLDER/bin/pip install --upgrade pip
+$VENV_FOLDER/bin/pip install -r requirements.txt
+$VENV_FOLDER/bin/pip install -r test-requirements.txt
+$PYTHON setup.py install
+git grep -l '2018-02' | xargs sed -i 's/2018-02/2021-02/g'
+
+if [ ! -d {{ ardana_qe_tests_dir }}/.testrepository ]; then
+    ${VENV_FOLDER}/bin/testr init
+fi
+
+cat >{{ ardana_qe_tests_dir }}/etc/functional_tests.conf <<EOF
+[DEFAULT]
+
+[identity]
+uri={{ os_unversioned_auth_url }}
+auth_version=v3
+username={{ os_admin_osrc.OS_USERNAME }}
+tenant_name=admin
+password={{ os_admin_osrc.OS_PASSWORD }}
+domain_name={{ os_admin_osrc.OS_DOMAIN_NAME }}
+admin_username={{ os_admin_osrc.OS_USERNAME }}
+admin_tenant_name=admin
+admin_password={{ os_admin_osrc.OS_PASSWORD }}
+admin_domain_name={{ os_admin_osrc.OS_DOMAIN_NAME }}
+
+[keymanager]
+url={{ km_url }}
+username={{ km_username }}
+password={{ km_password }}
+project_name={{ km_project_name }}
+project_id={{ km_project_id }}
+#max_payload_size=10000
+project_domain_name={{ km_project_domain_name }}
+
+EOF
+
+
+cat >{{ ardana_qe_tests_dir }}/.testr.conf <<EOF
+[DEFAULT]
+test_command=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
+             OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
+             OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
+             OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
+             ${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./functionaltests}
+test_list_option=--list
+group_regex=([^\.]*\.)*
+
+EOF
+
+cp /var/lib/ardana/openstack/ardana/ansible/roles/tempest/files/tests2skip.py {{ ardana_qe_tests_dir }}/tests2skip.py
+
+http_proxy= https_proxy= HTTP_PROXY= HTTPS_PROXY= ${TESTR} run \
+    --subunit $($PYTHON ${TESTS_TO_SKIP_BIN} ${RUN_FILTER})|
+    ${VENV_FOLDER}/bin/subunit-trace --no-failure-debug -f | tee testr_results.log
```

Comment:
Please save the test output to `{{ ardana_qe_test_log }}` so jenkins can store the test log.